src/test/s3/util/fileViewerManager.test.ts

```diff
@@ -3,6 +3,135 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import * as assert from 'assert'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { S3BucketNode } from '../../../s3/explorer/s3BucketNode'
+import { S3FileNode } from '../../../s3/explorer/s3FileNode'
+import { S3FolderNode } from '../../../s3/explorer/s3FolderNode'
+import { readablePath } from '../../../s3/util'
+import { S3FileViewerManager } from '../../../s3/util/fileViewerManager'
+import { DefaultFile, S3Client } from '../../../shared/clients/s3Client'
+import { makeTemporaryToolkitFolder } from '../../../shared/filesystemUtilities'
+import { getStringHash } from '../../../shared/utilities/textUtilities'
+import { FakeCommands } from '../../shared/vscode/fakeCommands'
+import * as testutil from '../../testUtil'
+import { anything, instance, mock, when } from '../../utilities/mockito'
+
 describe('FileViewerManager', function () {
-    //TODOD:: Not big enought to test yet
+    const bucketName = 'bucket-name'
+    const key = 'file.jpg'
+    const sizeBytes = 16
+    const creationDate = new Date(2020, 7, 7)
+    let file: DefaultFile
+    let testNode: S3FileNode
+    let s3: S3Client
+    let fileViewerManager: S3FileViewerManager
+    let window: typeof vscode.window
+    let testCache: Set<string>
+    let tempPath: string
+    let parent: S3BucketNode | S3FolderNode
+    let commands: FakeCommands
+    let tempFile: vscode.Uri
+
+    beforeEach(async function () {
+        window = mock()
+        s3 = mock()
+        parent = mock()
+        tempPath = await makeTemporaryToolkitFolder()
+        testCache = new Set<string>()
+        commands = new FakeCommands()
+        fileViewerManager = new S3FileViewerManager(testCache, instance(window), commands, tempPath)
+        file = new DefaultFile({
+            partitionId: 'aws',
+            bucketName,
+            key,
+            lastModified: creationDate,
+            sizeBytes,
+        })
+        testNode = new S3FileNode({} as any, file, instance(parent), instance(s3))
+
+        let completePath = getStringHash(readablePath(testNode))
+        //let completePath = readablePath(testNode).slice(4).split('/').join('%')
+        completePath = path.join(tempPath, 'S3%' + completePath)
+        tempFile = vscode.Uri.file(completePath)
+        testutil.toFile('bogus', tempFile.fsPath)
+    })
+
+    describe('retrieves file from s3 if not in temp or invalid date', function () {
```

Comment:
This and the `describe` wrapping this need to use `async function ()` since they wrap async functions.

Reply:
Eh not technically true. You only need `async` if you use `await` but that's kind of nit-picky.